src/libraries/System.IO.Compression.ZipFile/tests/ZipFile.Extract.cs

```diff
@@ -209,5 +216,2406 @@ public async Task DirectoryEntryWithData(bool async)
+            if (File.Exists(zipPath)) File.Delete(zipPath);
+
+            const string encPw = "EncOnly123!";
+            const ZipArchiveEntry.EncryptionMethod enc = ZipArchiveEntry.EncryptionMethod.ZipCrypto;
+
+            var encryptedItems = new (string Name, string Content)[]
+            {
+            ("secure/one.txt", "top-secret-1"),
+            ("secure/two.txt", "top-secret-2"),
+            };
+
+            var plainItems = new (string Name, string Content)[]
+            {
+            ("public/a.txt", "visible-a"),
+            ("public/b.txt", "visible-b"),
+            };
+
+            // Act 1: Create archive with both encrypted and plain entries
+            using (var za = ZipFile.Open(zipPath, ZipArchiveMode.Create))
+            {
+                // Encrypted
+                foreach (var it in encryptedItems)
+                {
+                    var entry = za.CreateEntry(it.Name);
+                    using var w = new StreamWriter(entry.Open(encPw, enc), Encoding.UTF8, bufferSize: 1024, leaveOpen: false);
+                    w.Write(it.Content);
+                }
+
+                // Plain
+                foreach (var it in plainItems)
+                {
+                    var entry = za.CreateEntry(it.Name); // default: no encryption
+                    using var w = new StreamWriter(entry.Open(), Encoding.UTF8, bufferSize: 1024, leaveOpen: false);
+                    w.Write(it.Content);
+                }
+            }
+
+            // Act 2: Read back—encrypted need password, plain do not
+            using (var za = ZipFile.Open(zipPath, ZipArchiveMode.Read))
+            {
+                // Encrypted
+                foreach (var it in encryptedItems)
+                {
+                    var e = za.GetEntry(it.Name);
+                    Assert.NotNull(e);
+                    using var r = new StreamReader(e!.Open(encPw), Encoding.UTF8, detectEncodingFromByteOrderMarks: true);
+                    string content = await r.ReadToEndAsync();
+                    Assert.Equal(it.Content, content);
+                }
+
+                // Plain
+                foreach (var it in plainItems)
+                {
+                    var e = za.GetEntry(it.Name);
+                    Assert.NotNull(e);
+                    using var r = new StreamReader(e!.Open(), Encoding.UTF8, detectEncodingFromByteOrderMarks: true);
+                    string content = await r.ReadToEndAsync();
+                    Assert.Equal(it.Content, content);
+
+                    // Ensure opening a plain entry with a password is rejected (or simply ignored depending on API)
+                    Assert.ThrowsAny<Exception>(() =>
+                    {
+                        using var _ = e.Open("some-password");
+                    });
+                }
+            }
+        }
+
+        [Fact]
+        public async Task ZipCrypto_AsyncWrite_ThenAsyncRead_ContentMatches()
```

Comment:
Testing of sync and async methods can be deduplicated, see e.g. tests in zip_InvalidParametersAndStrangeFiles. They are using helpers from the base class and parameterize tests on a boolean argument whether async path should be tested